test/header/CMakeLists.txt

```diff
@@ -102,17 +97,18 @@ macro (seqan3_header_test component header_base_path)
 TEST(Header, ${header_source_test_name}) {}")
 
             add_library (${target_library} OBJECT "${source_file}")
-            target_compile_options (${target_library} PRIVATE ${cxx_flags_list} ${SEQAN3_TEST_CXX_FLAGS})
-            target_compile_definitions (${target_library} PRIVATE ${SEQAN3_DEFINITIONS} ${SEQAN3_TEST_DEFINITIONS})
-            target_include_directories (${target_library} PRIVATE ${SEQAN3_INCLUDE_DIRS} ${SEQAN3_TEST_INCLUDE_DIRS})
+            # NOTE: a simple target_link_libraries (${target_library} seqan3_unit_test)
+            # is not possible for OBJECT libraries
```

Comment:
whats an "object library"?

Reply:
It is a pseudo-library. If you add sources to `add_executable(target a.cpp b.cpp)` when internally you will get for each `.cpp` file a `.o` file. The problem is if you change any header/source file you have to rebuild every `.cpp` file of the `target`. What I do here is to build the `.o` for each `.cpp` individually. Such that a source change will only affect a subset off all `.cpp` files.